tests/end2end/features/test_completion_bdd.py

```diff
@@ -0,0 +1,29 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015-2016 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+import re
+import pytest_bdd as bdd
+bdd.scenarios('completion.feature')
+
+
+@bdd.then(bdd.parsers.parse("the completion model should be {model}"))
```

Comment:
While I'd like to have end-to-end tests for completion, I don't like the idea of relying on a debug message to prove that it is working properly. I tried writing completion feature tests that did things like `:open <Tab> <Tab> <Enter>` and made sure it ended up on the right page, but they seemed flaky.

For now, I'd rely on the unit tests. [`test_update_completion`](https://github.com/The-Compiler/qutebrowser/blob/master/tests/unit/completion/test_completer.py#L196) should do what you want.

Just make sure to change the signature of the [fake `set` command](https://github.com/The-Compiler/qutebrowser/blob/master/tests/unit/completion/test_completer.py#L98)